### PR TITLE
refactor(core): Create 'paths' package for better file paths

### PR DESCRIPTION
**Pull request by timoffex**
Description
---
I've spent too much time documenting whether paths are absolute, relative, slash-separated, etc. and debugging issues when they're not. This package makes it clear from the type signature.

I update `runfiles` and `tensorboard` to use these.

### CODE REVIEW

File: core/internal/paths/paths.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+	return toPtr(AbsolutePath(absPath)), nil
+}
+
+// Relative returns the path if it's relative, or returns an error.
+func Relative(path string) (*RelativePath, error) {
+	if filepath.IsAbs(path) {
+		return nil, fmt.Errorf("path is not relative: %s", path)
+	} else {
+		return toPtr(RelativePath(filepath.Clean(path))), nil
+	}
```
Comment (kptkin) [inline core/internal/paths/paths.go:79]: nit: no need for else, already have a return

File: core/internal/tensorboard/tfeventreader.go [hunk 2]
```diff
@@ -120,22 +121,22 @@ func (s *TFEventReader) NextEvent(
...
 	if uint64(len(s.buffer)) >= count {
 		return true
 	}
 
 	// If we haven't found the first file, try to find it.
-	if s.currentFile == "" {
+	if s.currentFile == nil {
 		var err error
-		s.currentFile, err = nextTFEventsFile(s.logDir, "", s.fileFilter)
+		s.currentFile, err = nextTFEventsFile(s.logDir, nil, s.fileFilter)
```
Comment (kptkin) [inline core/internal/tensorboard/tfeventreader.go:133]: nit: could you add a comment for the `nil` argument?

**Review by kptkin — APPROVED** [review]
Nice! this is a constant source of confusion. 
 Rel vs Abs path for our different operations

File: core/internal/tensorboard/tfeventreader.go (hunk 2 above)
Comment (timoffex) [reply core/internal/tensorboard/tfeventreader.go]: I created a `nextTFEventsFile` method that takes no arguments instead :)

Comment (kptkin) [reply core/internal/tensorboard/tfeventreader.go]: Ah didn’t get there yet… noce

**Pull request merged**